tests/integration/v3_leadership_test.go

```diff
@@ -179,3 +180,77 @@ func TestTransferLeadershipWithLearner(t *testing.T) {
 		t.Error("timed out waiting for leader transition")
 	}
 }
+
+func TestFirstCommitNotification(t *testing.T) {
+	BeforeTest(t)
+
+	clusterSize := 3
+	cluster := NewClusterV3(t, &ClusterConfig{Size: clusterSize})
+	defer cluster.Terminate(t)
+
+	oldLeaderIdx := cluster.WaitLeader(t)
+	oldLeaderClient := cluster.Client(oldLeaderIdx)
+
+	newLeaderIdx := (oldLeaderIdx + 1) % clusterSize
+	newLeaderId := uint64(cluster.Members[newLeaderIdx].ID())
+
+	notifiers := make(map[int]<-chan struct{}, clusterSize)
+	for i, clusterMember := range cluster.Members {
+		notifiers[i] = clusterMember.s.FirstCommitInTermNotify()
+	}
+
+	_, err := oldLeaderClient.MoveLeader(context.Background(), newLeaderId)
+
+	if err != nil {
+		t.Errorf("got error during leadership transfer: %v", err)
+	}
+
+	leaderAppliedIndex := cluster.Members[newLeaderIdx].s.AppliedIndex()
+	doneC := make(chan string, clusterSize)
+
+	for i, notifier := range notifiers {
+		go func(member *member, notifier <-chan struct{}) {
+			timeout := time.NewTimer(5 * time.Second)
+			defer timeout.Stop()
+			// wait until server applies all the changes of leader
+			for member.s.AppliedIndex() < leaderAppliedIndex {
+				select {
+				case <-timeout.C:
+					doneC <- fmt.Sprintf(
+						"member with ID %d failed to apply new entries",
+						member.ID(),
+					)
+					return
+				default:
+					time.Sleep(100 * time.Millisecond)
+				}
+			}
+
+			select {
+			case msg, ok := <-notifier:
+				if ok {
+					doneC <- fmt.Sprintf(
+						"member with ID %d got message via notifier, msg: %v",
```

Comment:
fmt.Errorf would make it more obvious that its the error...

but than we have problem how to 'send' success (""), 

and actually Group.Wait mitigates that dilema (see below).